src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,36 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { Todo } from '../../types/Todo';
+
+interface TodoModalProps {
+  todo: Todo;
+  setSelectedTodoId: React.Dispatch<React.SetStateAction<number>>;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  setSelectedTodoId,
+  todo,
+}) => {
+  const [isLoading, setIsLoading] = useState(true);
+  const [user, setUser] = useState<User>();
+
+  useEffect(() => {
+    getUser(todo.userId)
+      .then(userData => {
+        setUser(userData);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  }, [todo.userId]);
```

Comment:
You are initializing your state with props in line 17 which can lead to potential bugs in your application. Here are some resources to help you understand why this can be problematic: [Resource 1](https://stackoverflow.com/a/50403930), [Resource 2](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state), [Resource 3](https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state)